www/js/shortcuts/shortcut-functions.js

```diff
@@ -0,0 +1,166 @@
+  $line.click();
+  maintainScroll($line);
+};
+
+// is the current slide of the shabad (= false) or an inserted slide (= true)
+let isInsertedSlide = false;
+const updateInsertedSlide = newValue => {
+  isInsertedSlide = newValue;
+};
+
+/* ******************
+ * SLIDE SHORTCUTS
+ ******************** */
+const waheguru = () => {
+  global.controller.sendText(strings.slideStrings.waheguru, true);
+  isInsertedSlide = true;
+};
+const empty = () => {
+  global.controller.sendText(' ');
+  isInsertedSlide = true;
+};
+const moolMantra = () => {
+  global.controller.sendText(strings.slideStrings.moolMantra, true);
+  isInsertedSlide = true;
+};
+
+/* *******************
+ * CEREMONY SHORTCUTS
+ ******************** */
+const anandSahibBhog = () => {
+  global.core.search.loadCeremony(3).catch(error => {
+    analytics.trackEvent('ceremonyFailed', 3, error);
+  });
+};
+
+/* *******************
+ * INTERFACE SHORTCUTS
+ ******************** */
+const help = () => main.openSecondaryWindow('helpWindow');
+const legend = () => main.openSecondaryWindow('shortcutLegend');
+const searchBar = () => {
+  search.$search.focus();
+  search.$search.value = '';
+};
+
+/* *******************
+ * OTHER SHORTCUTS
+ ******************** */
+const openFirstResult = () => {
+  if (document.activeElement.id === 'search') {
+    document.querySelector('#results .search-result').click();
+    document.getElementById('shabad-page').focus();
+  }
+};
+
+const spaceBar = e => {
+  const currentLineId = search.$shabad.querySelector('a.panktee.current').dataset.lineId;
+  if (!isInsertedSlide) {
+    const mainLineID = search.$shabad.querySelector('a.panktee.main').dataset.lineId;
+
+    let newLineId = mainLineID;
+
+    if (mainLineID === currentLineId) {
+      newLineId = search.$shabad.querySelector('a.panktee:not(.seen_check)').dataset.lineId;
+    }
+
+    highlightLine(newLineId);
+    e.preventDefault();
+  } else {
+    highlightLine(currentLineId);
+    document.getElementById('shabad-page').focus();
+    isInsertedSlide = false;
+  }
+};
+
+const prevLine = e => {
+  // Find selector of current line in Shabad
+  const $currentLine = search.$shabad.querySelector('a.panktee.current').parentNode;
+  const $prevLine = $currentLine.previousElementSibling;
+  // if its not at the topmost panktee
+  if ($prevLine && $prevLine.dataset.lineCount) {
+    const $prevPanktee = $prevLine.querySelector('a.panktee');
+    $prevPanktee.click();
+    maintainScroll($prevPanktee);
+  }
+  e.preventDefault();
+};
+
+const nextLine = e => {
+  // Find selector of current line in Shabad
+  const $currentLine = search.$shabad.querySelector('a.panktee.current').parentNode;
+  const $nextLine = $currentLine.nextElementSibling;
+  // if its not at the last panktee
+  if ($nextLine && $nextLine.dataset.lineCount) {
+    const $nextPanktee = $nextLine.querySelector('a.panktee');
+    $nextPanktee.click();
+    maintainScroll($nextPanktee);
+  }
+  e.preventDefault();
```

Comment:
this is throwing errors 
```
sttm-desktop/www/js/shortcuts/shortcut-functions.js:134 
Uncaught TypeError: Cannot read property 'preventDefault' of undefined
```

Reply:
I don't see any errors now, probably because of mousetrap? And viewer is visible to me. 🤷‍♂

Reply:
Couldn't reproduce.

Reply:
still seeing it. see slack.